gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/CsvOptions.java

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.bigquery;
+
+import com.google.common.base.MoreObjects;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Google BigQuery CSV options. This class wraps some properties of CSV files used by BigQuery to
+ * parse external data.
+ */
+public class CsvOptions implements Serializable {
+
+  private static final long serialVersionUID = 2193570529308612708L;
+
+  private final Boolean allowJaggedRows;
+  private final Boolean allowQuotedNewLines;
+  private final String encoding;
+  private final String fieldDelimiter;
+  private final String quote;
+  private final Integer skipLeadingRows;
+
+  public static final class Builder {
+
+    private Boolean allowJaggedRows;
+    private Boolean allowQuotedNewLines;
+    private String encoding;
+    private String fieldDelimiter;
+    private String quote;
+    private Integer skipLeadingRows;
+
+    private Builder() {}
+
+    /**
+     * Set whether BigQuery should accept rows that are missing trailing optional columns. If
+     * {@code true}, BigQuery treats missing trailing columns as null values. If {@code false},
+     * records with missing trailing columns are treated as bad records, and if there are too many
+     * bad records, an invalid error is returned in the job result. By default, rows with missing
+     * trailing columns are considered bad records.
+     */
+    public Builder allowJaggedRows(Boolean allowJaggedRows) {
+      this.allowJaggedRows = allowJaggedRows;
+      return this;
+    }
+
+    /**
+     * Sets whether BigQuery should allow quoted data sections that contain newline characters in a
+     * CSV file. By default quoted newline are not allowed.
+     */
+    public Builder allowQuotedNewLines(Boolean allowQuotedNewLines) {
+      this.allowQuotedNewLines = allowQuotedNewLines;
+      return this;
+    }
+
+    /**
+     * Sets the character encoding of the data. The supported values are UTF-8 or ISO-8859-1. The
+     * default value is UTF-8. BigQuery decodes the data after the raw, binary data has been split
+     * using the values set in {@link #quote(String)} and {@link #fieldDelimiter(String)}.
+     */
+    public Builder encoding(String encoding) {
```

Comment:
Fine as is. Alternative (or addition) could be to accept a `Charset` and then use its `name()`